src/modules/visits/services/visits.service.ts

```diff
@@ -250,11 +257,40 @@ export class VisitsService {
 			}
 		}
 
+		if (updateVisit.access_type) {
+			// IF accessFolderIds is empty and accessType is FOLDERS
+			// OR accessFolderIds is not empty and accessType is FULL
+			// THEN throw BadRequest exception
+			if (
+				isEmpty(accessType) ||
+				(isEmpty(accessFolderIds) && updateVisit.access_type === VisitAccessType.FOLDERS) ||
+				(!isEmpty(accessFolderIds) && updateVisit.access_type === VisitAccessType.FULL)
+			) {
+				throw new BadRequestException(
+					`The amount of accessFolderIds (${accessFolderIds.length}) does not correspond with the accessType ${updateVisit.access_type}`
+				);
+			}
+
+			// IF accessFolderIds is not empty and accessType is FOLDERS
+			// THEN add these folders to the folder_access table
+			if (!isEmpty(accessFolderIds) && updateVisit.access_type === VisitAccessType.FOLDERS) {
+				for (let index = 0; index < accessFolderIds.length; index++) {
+					await this.dataService.execute<
+						InsertVisitFolderAccessMutation,
+						InsertVisitFolderAccessMutationVariables
+					>(InsertVisitFolderAccessDocument, {
+						folderId: accessFolderIds[index],
+						visitRequestId: currentVisit.id,
+					});
+				}
```

Comment:
If 10 folders are selected, this code will do 10 round trips to the database before completing, which is quite slow.

You can modify the query to allow to insert multiple entries at once:
```graphql
mutation insertVisitFolderAccess($objects: [maintainer_visitor_space_request_folder_access_insert_input!] = {}) {
  insert_maintainer_visitor_space_request_folder_access(objects: $objects) {
    returning {
      id
    }
  }
}
```

There are 3 options:
* current solution: (slow)
	* make first call to db
	* receive response
	* make second call to db
	* receive response
	* make third call to db
	* receive response
	* make fourth call to db
	* receive response
* option with promise.all (fast but a lot of load on the db)
	* make first, second, third, fourth call to db
	* receive responses
* option with modified query: (fast and not a lot of load on the db)
	* make one call to the db with item 1, 2, 3, 4
	* receive response ids for 1, 2, 3, 4